plugins/ltac/tactic_debug.ml

```diff
@@ -163,20 +163,20 @@ let cvt_loc loc =
          let file = String.sub dp (lastdot+1) (dplen - (lastdot + 1)) in
          let module_name = String.sub dp 0 lastdot in
          let routine =
-           try
-             (* try text as a kername *)
-             assert (CString.is_prefix dp text);
-             let knlen = String.length text in
-             let lastdot = String.rindex text '.' in
-             String.sub text (lastdot+1) (knlen - (lastdot + 1))
-           with exn when CErrors.noncritical exn -> text
+           (* try text as a kername *)
+           if not (CString.is_prefix dp text) then text else
+             try
+               let knlen = String.length text in
+               let lastdot = String.rindex text '.' in
+               String.sub text (lastdot+1) (knlen - (lastdot + 1))
+             with _ -> text
```

Comment:
What is this trying to achieve exactly? Barring asynchronous exceptions the only pieces of code there that can raise are String.rindex and String.sub and it's easy to check you're within bounds.

Reply:
If there's no dot, there's no path prefix to remove, so just use 'text' as-is.  I expect `rindex` can raise `Not_found` even if the manual doesn't say so (like `rindex_from`).